src/main/java/net/minecraftforge/items/wrapper/SidedInvWrapper.java

```diff
@@ -0,0 +1,184 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+public class SidedInvWrapper implements IItemHandlerModifiable
+{
+    protected final ISidedInventory inv;
+    protected final EnumFacing side;
+
+    public SidedInvWrapper(ISidedInventory inv, EnumFacing side)
+    {
+        this.inv = inv;
+        this.side = side;
+    }
+
+    public static int getSlot(ISidedInventory inv, int slot, EnumFacing side)
+    {
+        int[] slots = inv.getSlotsForFace(side);
+        if (slot < slots.length)
+            return slots[slot];
+        return -1;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+
+        SidedInvWrapper that = (SidedInvWrapper) o;
+
+        return inv.equals(that.inv) && side == that.side;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = inv.hashCode();
+        result = 31 * result + side.hashCode();
+        return result;
+    }
+
+    @Override
+    public int getSlots()
+    {
+        return inv.getSlotsForFace(side).length;
+    }
+
+    @Override
+    public ItemStack getStackInSlot(int slot)
+    {
+        int i = getSlot(inv, slot, side);
+        return i == -1 ? null : inv.getStackInSlot(i);
+    }
+
+    @Override
+    public ItemStack insertItem(int slot, ItemStack stack, boolean simulate)
+    {
+
+        if (stack == null)
+            return null;
+
+        int slot1 = getSlot(inv, slot, side);
+
+        if (slot1 == -1)
+            return stack;
+
+        if (!inv.isItemValidForSlot(slot1, stack) || !inv.canInsertItem(slot1, stack, side))
+            return stack;
+
+        ItemStack stackInSlot = inv.getStackInSlot(slot1);
+
+        int m;
+        if (stackInSlot != null)
+        {
+            if (!ItemHandlerHelper.canItemStacksStack(stack, stackInSlot))
+                return stack;
+
+            m = Math.min(stack.getMaxStackSize(), inv.getInventoryStackLimit()) - stackInSlot.stackSize;
```

Comment:
`m` is not a good variable name. Especially if its still used 20 lines after its initialization.

Also it looks like its used for 2 purposes. Split it into 2 variables with proper names.